src/accelerate/data_loader.py

```diff
@@ -366,13 +367,23 @@ class DataLoaderShard(DataLoader, DataLoaderStateMixin):
         - **total_dataset_length** (`int`) -- Total length of the inner dataset across all processes.
     """
 
-    def __init__(self, dataset, device=None, rng_types=None, synchronized_generator=None, skip_batches=0, **kwargs):
+    def __init__(
+        self,
+        dataset,
+        device=None,
+        rng_types=None,
+        synchronized_generator=None,
+        skip_batches=0,
+        _drop_last: bool = False,
```

Comment:
Why use an underscore for the argument name? That's quite unusual. `drop_last` on the `dataloader` doesn't use an underscore. If the attribute should be marked as "private", it can still use an underscore, but for the argument name, I'd go without (i.e. `self._drop_last = drop_last`).

Reply:
Because it is indeed private, and to be consistent with the other dataloader arg which has it as private too. Because we control it, the user shouldn't ever pass in `_drop_last` (but we need to on the backend)

Reply:
I see. It's a bit strange to the eye, but as you said, it should not be visible to the user.